File: javascript/src/challenges/same_frequency.test.ts
import sameFrequency from './same_frequency'

describe('Same Frequency Challenge', () => {
    it('should return false when input has diff length', () => {
        const num1 = 22
        const num2 = 222
        const result = sameFrequency(num1, num2)
        expect(result).toBeFalsy()
    })

    it('should return false when length is same but frequency is wrong', () => {
        const num1 = 34
        const num2 = 14
        const result = sameFrequency(num1, num2)
        expect(result).toBeFalsy()
    })

    it('should return true when length and frequency is same with small numbers', () => {
        const num1 = 182
        const num2 = 281
        const result = sameFrequency(num1, num2)
        expect(result).toBeTruthy()
    })

    it('should return true when length and frequency is same with large numbers', () => {
        const num1 = 3589578
        const num2 = 5879385
        const result = sameFrequency(num1, num2)
        expect(result).toBeTruthy()
    })
})

File: go/challenges/same_frequency_test.go
package challenges

import "testing"

func TestSameFrequency(t *testing.T) {
	t.Run("should return false if the len of the num is diff", func(t *testing.T) {
		num1 := 22
		num2 := 222
		got := SameFrequency(num1, num2)
		want := false
		if got != want {
			t.Errorf("expected %v got %v", got, want)
		}
	})

	t.Run("should return false when some has not same frequency of digs", func(t *testing.T) {
		num1 := 34
		num2 := 14
		got := SameFrequency(num1, num2)
		want := false
		if got != want {
			t.Errorf("expected %v got %v", got, want)
		}
	})

	t.Run("should return true when short number has same frequency of digs", func(t *testing.T) {
		num1 := 182
		num2 := 281
		got := SameFrequency(num1, num2)
		want := true
		if got != want {
			t.Errorf("expected %v got %v", got, want)
		}
	})

	t.Run("should return true when large number has same frequency of digs", func(t *testing.T) {
		num1 := 3589578
		num2 := 5879385
		got := SameFrequency(num1, num2)
		want := true
		if got != want {
			t.Errorf("expected %v got %v", got, want)
		}
	})
}
